packages/supi/src/api/storeAdd.ts

```diff
@@ -0,0 +1,64 @@
+import normalizeRegistryUrl = require('normalize-registry-url')
+
+import logger, {streamParser} from '@pnpm/logger'
+import {StoreController} from 'package-store'
+
+import parseWantedDependencies from '../parseWantedDependencies';
+import {ReporterFunction} from '../types'
+
+export default async function (
+  fuzzyDeps: string[],
+  opts: {
+    registry?: string,
+    verifyStoreIntegrity?: boolean,
+    reporter?: ReporterFunction,
+    storeController: StoreController,
+    tag?: string,
+  },
+) {
+  const reporter = opts && opts.reporter
+  if (reporter) {
+    streamParser.on('data', reporter)
+  }
+
+  const deps = await parseWantedDependencies(fuzzyDeps, {
+    allowNew: true,
+    currentPrefs: {},
+    defaultTag: opts.tag || 'latest',
+    dev: false,
+    devDependencies: {},
+    optional: false,
+    optionalDependencies: {},
+  })
+
+  let hasFailures = false;
+  await Promise.all(deps.map(async (dep) => {
+    try {
+      const pkgResponse = await opts.storeController.requestPackage(dep, {
+        downloadPriority: 1,
+        loggedPkg: {
+          rawSpec: dep.raw,
+        },
+        preferredVersions: {},
+        prefix: '',
```

Comment:
should it work with local tarballs? If you pass in the prefix, it will work with `pnpm store add ../some.tgz`

Reply:
I have no idea what the prefix is

Reply:
it should be passed in by pnpm. prefix is basically the current working directory in most cases